src/structures/textChannel.ts

```diff
@@ -76,9 +76,54 @@ export class TextChannel extends Channel {
       }
     )
 
-    // TODO: Actually construct this object
-    const mentions = new MessageMentions()
+    const res = new Message(this.client, newMsg, this, this.client.user)
+    await res.mentions.fromPayload(newMsg)
+    return res
+  }
+}
+
+export class GuildTextChannel extends TextChannel {
```

Comment:
Why did you merge GuildTextChannel into TextChannel?

Reply:
It was throwing unexpected error saying "cannot access TextChannel before initialization" in guildTextChannel.ts while it was just exported one. I merged it there and it worked.

Reply:
I think it's due to the import cycle, and no better way to fix it now. I'll merge it now.